syntax = "proto3";

//Пара вспомогательных типов данных 

enum MorphMsg {
    ANY = 0;
    UNKN = 1;
    VERB = 2;
    NOUN = 3;
    PROPN = 4;
    INFN = 5;
    PRTF = 6;
}

message WordMsg {
    string text = 1;
    MorphMsg morph = 2;
}

message SimilarWordMsg {
    WordMsg word = 1;
    double similarity = 2;
}

//Запросы и ответы

message WordInitRequest {
    MorphMsg morph = 1;
    int32 similar_words_count = 2;
}

message WordInitResponse {
    WordMsg word = 1;
    repeated SimilarWordMsg similar_words = 2;
}

message SimilarityRequest {
    WordMsg original_word = 1;
    WordMsg test_word = 2;
}

message SimilarityResponse {
    SimilarWordMsg word = 1;
}

//Микросервис собственной персоной

service MorphModelService {
    rpc WordInit(WordInitRequest) returns (WordInitResponse);
    rpc Similarity(SimilarityRequest) returns (SimilarityResponse);
}